mon_school/mon_school/doctype/lms_sketch_like/lms_sketch_like.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2021, FOSS United and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+
+class LMSSketchLike(Document):
+    def after_insert(self):
+        sketch = self.get_sketch()
+        sketch.update_metrics()
+
+    def get_sketch(self):
+        return frappe.get_doc("LMS Sketch", self.sketch)
+
+
+def on_doctype_update():
+    # only one like for each sketch/user
+    frappe.db.add_unique("LMS Sketch Like", fields=["sketch", "user"])
+
+    # index for faster queries
+    frappe.db.add_index("LMS Sketch Like", fields=["sketch"])
+    frappe.db.add_index("LMS Sketch Like", fields=["user"])
```

Comment:
Index can be added from desk. It gets recorded in the JSON file of the doctype. Using that would be a good idea instead of writing this code.

Reply:
I couldn't find it on desk. Can you share some documentation/code reference for this?

Reply:
<img width="1676" alt="Screenshot 2021-12-23 at 3 31 29 PM" src="https://user-images.githubusercontent.com/7569/147224340-fb9f5951-36bb-423c-bb3d-7fdc2a2544e0.png">

Reply:
Ah, the forms are different for `/app/doctype/<doctype-name>` and `/app/customize-form`.
Found it :+1:

Reply:
Done :+1: